components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/OauthAuthenticator.java

```diff
@@ -0,0 +1,170 @@
+
+    private static final Log log = LogFactory.getLog(OauthAuthenticator.class);
+    private static final int PRIORITY = 5;
+    private static final int ACCESS_TOKEN_INDEX = 1;
+
+    private static String hostUrl = "";
+    private static boolean isRemote = false;
+
+    static {
+        AuthenticatorsConfiguration authenticatorsConfiguration = AuthenticatorsConfiguration.getInstance();
+        AuthenticatorsConfiguration.AuthenticatorConfig authenticatorConfig = authenticatorsConfiguration.getAuthenticatorConfig(OauthAuthenticatorConstants.AUTHENTICATOR_NAME);
+
+        if (authenticatorConfig != null) {
+            isRemote = Boolean.parseBoolean(authenticatorConfig.getParameters().get("isRemote"));
+            hostUrl = authenticatorConfig.getParameters().get("hostURL");
+
+        }
+    }
+
+    /**
+     * Checks whether the authentication of the context can be handled using this authenticator.
+     *
+     * @param messageContext containing the request need to be authenticated.
+     * @return boolean indicating whether the request can be authenticated by this Authenticator.
+     */
+    public boolean isHandle(MessageContext messageContext) {
+        HttpServletRequest httpServletRequest = getHttpRequest(messageContext);
+        String headerValue = httpServletRequest.getHeader(HTTPConstants.HEADER_AUTHORIZATION);
+
+        if (headerValue != null && !headerValue.trim().isEmpty()) {
+            String[] headerPart = headerValue.trim().split(OauthAuthenticatorConstants.SPLITING_CHARACTOR);
+
+            if (OauthAuthenticatorConstants.AUTHORIZATION_HEADER_PREFIX_BEARER.equals(headerPart[0])) {
+                return true;
+            }
+        } else if (httpServletRequest.getParameter(OauthAuthenticatorConstants.BEARER_TOKEN_IDENTIFIER) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Authenticates the user using the provided OAuth token and returns the status as a boolean.
+     * Sets the tenant domain and tenant friendly username to the session as attributes.
+     *
+     * @param messageContext containing the request need to be authenticated.
+     * @return boolean indicating the authentication status.
+     */
+    public boolean isAuthenticated(MessageContext messageContext) {
+        HttpServletRequest httpServletRequest = getHttpRequest(messageContext);
+        String headerValue = httpServletRequest.getHeader(HTTPConstants.HEADER_AUTHORIZATION);
+        //split the header value to separate the identity type and the token.
+        String[] headerPart = headerValue.trim().split(OauthAuthenticatorConstants.SPLITING_CHARACTOR);
+        String accessToken = headerPart[ACCESS_TOKEN_INDEX];
+        OAuth2TokenValidator tokenValidator = OAuthValidatorFactory.getValidator(isRemote,hostUrl);
+
+        if (tokenValidator == null) {
+            log.error("OAuthValidationFactory failed to return a validator",
```

Comment:
Log and throw is usually considered an anti-pattern. Therefore, just throwing the exception would do.